data/tutorials/language/0it_05_imperative.md

```diff
@@ -526,6 +526,18 @@ The following example uses the `get_char` function we defined earlier (in the se
 
 This `while` loop echoes characters typed on the keyboard. When the ASCII `Escape` character is read, the `Exit` exception is thrown, which terminates the iteration and displays the REPL reply: `- : unit = ()`.
 
+### Closures
+
+A closure is a function bundled together with its surrounding state, including any variables or references from the enclosing scope that it references. This bundled function-state package allows for the creation of self-contained units of behavior that can maintain internal state across multiple invocations. 
+
+```ocaml
+let counter () =
+  let n = ref 0 in
+  fun () -> incr n; !n;;
+```
```

Comment:
We've been formatting toplevel code examples with a `#` at the beginning and the expected return beneath the code. Shall I change it to match?

Reply:
Oh, I see. Thanks for clarifying. :smiley: